src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -0,0 +1,266 @@
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(short value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(long value)
+        {
+            if (this.isIeee754Compatible)
+            {
+                this.writer.WriteStringValue(value.ToString(CultureInfo.InvariantCulture));
+            }
+            else
+            {
+                this.writer.WriteNumberValue(value);
+            }
+            
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(double value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(Guid value)
+        {
+            this.writer.WriteStringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(decimal value)
+        {
+            if (this.isIeee754Compatible)
+            {
+                this.writer.WriteStringValue(value.ToString(CultureInfo.InvariantCulture));
+            }
+            else
+            {
+                this.writer.WriteNumberValue(value);
+            }
+
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(DateTimeOffset value)
+        {
+            this.writer.WriteStringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(TimeSpan value)
+        {
+            string stringValue = EdmValueWriter.DurationAsXml(value);
+            this.writer.WriteStringValue(stringValue);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(Date value)
+        {
+            this.writer.WriteStringValue(value.ToString());
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(TimeOfDay value)
+        {
+            this.writer.WriteStringValue(value.ToString());
+        }
+
+        public void WriteValue(byte value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(sbyte value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(string value)
+        {
+            if (value == null)
+            {
+                this.writer.WriteNullValue();
+            }
+            else
+            {
+                this.writer.WriteStringValue(value);
+            }
+
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(byte[] value)
+        {
+            this.writer.WriteBase64StringValue(value);

```

Comment:
```suggestion
            if (value == null)
            {
                this.writer.WriteNullValue();
            }
            else
            {
                this.writer.WriteBase64StringValue(value);
            }
```

This change preserves the current behavior. As written, we would write "[]" if `value` is `null`.

Reply:
Thanks, I hadn't caught this, partially because of oversight, and partially because there wasn't an existing unit test covering this.